rnd/autogpt_server/autogpt_server/server/rest_api.py

```diff
@@ -58,25 +42,22 @@ def get_user_id(payload: dict = Depends(auth_middleware)) -> str:
 
 
 class AgentServer(AppService):
-    event_queue: asyncio.Queue[ExecutionResult] = asyncio.Queue()
-    manager = ConnectionManager()
     mutex = KeyedMutex()
     use_db = False
     _test_dependency_overrides = {}
 
-    async def event_broadcaster(self):
-        while True:
-            event: ExecutionResult = await self.event_queue.get()
-            await self.manager.send_execution_result(event)
+    def __init__(self, event_queue: AsyncEventQueue | None = None):
+        self.event_queue = event_queue or AsyncRedisEventQueue()
 
     @asynccontextmanager
     async def lifespan(self, _: FastAPI):
+        self.run_and_wait(self.event_queue.connect())
         await db.connect()
         await block.initialize_blocks()
         if await user_db.create_default_user(settings.config.enable_auth):
             await graph_db.import_packaged_templates()
-        asyncio.create_task(self.event_broadcaster())
         yield
+        self.run_and_wait(self.event_queue.close())
```

Comment:
Do we need to use `self.run_and_wait(..)`? The rest of `lifespan()` uses `await`.